pkg/parser/ast/visitor_codegen/generator.go

```
@@ -0,0 +1,1759 @@
+	return buffer.String()
+}
+
+// WriteFileAtomically writes generated output without exposing a partial file.
+func WriteFileAtomically(path string, contents []byte) error {
+	directory := filepath.Dir(path)
+	temporary, err := os.CreateTemp(directory, ".visitor-inplace-*.tmp")
+	if err != nil {
+		return err
+	}
+	temporaryName := temporary.Name()
+	defer os.Remove(temporaryName)
```

golangci-lint (2.12.2)

[error] Error return value of `os.Remove` is not checked

(errcheck)